### PR TITLE
Fix no cards loading

### LINKED ISSUE #68: Card don't opens when open direct link
Steps to reproduce:
1. Open https://test.cards.atomix.team/open/29
2. Wait
3. Just skeleton loading

Expected:
...
3. Card loaded and show

### LINKED ISSUE #69: Home page don't load cards when open direct
Steps to reproduce:
1. Open https://test.cards.atomix.team/
2. Wait
3. "Not found" text
4. Go to join
5. Go to home page
6. Cards loaded

Expected:
1. Open https://test.cards.atomix.team/
2. Wait
3. Cards loaded

### PR DESCRIPTION
**Pull request by sergeysova**
Closes #69 
Closes #68

### CODE REVIEW

**Comment by meowsun** [comment]
Looks elegant

File: src/features/cards/organisms/cards-list.js [hunk 1]
```diff
@@ -1,63 +1,58 @@
...
-        )}
-      </>
-    )
-  },
+  ({ renderCard = defaultCardRender, renderEmpty = emptyRenderer }, cards) => (
+    <ConditionalList
+      list={cards}
+      renderExists={(list) => (
+        <CardsItemsBlock>
+          {list.filter(Boolean).map((card) =>
```
Comment (artalar) [inline src/features/cards/organisms/cards-list.js]: `list.map(card => Boolean(card) && render...`

File: src/features/cards/organisms/skeleton-list.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  ids: number[],
+  count?: number,
+  renderEmpty: () => React.Node,
+  renderCard?: *,
+}
+
+export const SkeletonList = ({
+  isLoading,
+  ids,
+  count = 3,
```
Comment (artalar) [inline src/features/cards/organisms/skeleton-list.js]: magic numbers. Need to move to named const

File: src/features/cards/organisms/skeleton-list.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  renderEmpty,
+  renderCard,
+}: Props) =>
+  isLoading ? (
+    <>
```
Comment (artalar) [inline src/features/cards/organisms/skeleton-list.js]: `Fragment` is not needed

File: src/features/cards/organisms/skeleton-list.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      {Array.from({ length: count }, (_, idx) => (
+        <CardSkeleton key={idx} />
+      ))}
+    </>
+  ) : (
+    <CardsList ids={ids} renderEmpty={renderEmpty} renderCard={renderCard} />
+  )
+
+SkeletonList.defaultProps = {
+  count: undefined,
```
Comment (artalar) [inline src/features/cards/organisms/skeleton-list.js]: It doit nothing, why it needs?

File: src/features/cards/pages/view.js [hunk 5]
```diff
@@ -1,40 +1,51 @@
...
-            card={card}
-            onUsefulClick={onUsefulClick}
-          />
-        )}
-      />
+      {current ? (
+        <CardsList
+          key={`card-view-${current.id}`}
+          ids={[current.id]}
+          renderCard={({ card, onUsefulClick }) => (
```
Comment (artalar) [inline src/features/cards/pages/view.js]: That function can be static (defined in the upper scope)

File: src/features/users/pages/user.js [hunk 6]
```diff
@@ -41,72 +41,54 @@ export const UserPage = ({ match }: Props) => {
...
+        Useful
       </ZeroTab>
       |
-      <ZeroTab
-        onClick={() => {
-          setFilter("useful")
-        }}
-      >
-        Favorite
+      <ZeroTab active={tab === "created"} onClick={() => setTab("created")}>
```
Comment (artalar) [inline src/features/users/pages/user.js]: All string literals (`"useful"`, `"created"`) must be defined statically in one place and reused by imports for safety feature changes

File: src/features/users/pages/user.js [hunk 7]
```diff
@@ -150,28 +132,28 @@ CurrentUserInfo.propTypes = {
...
-      </>
-    )
-  }
-  return <Col padding="2rem">{renderEmpty()}</Col>
+const NamedCardsList = ({
+  show,
+  isLoading,
+  cards,
+  title,
+  renderEmpty = () => null,
```
Comment (artalar) [inline src/features/users/pages/user.js]: Duplicated from defaultProps. It is for better static type checking or what?

File: src/features/cards/organisms/cards-list.js [hunk 8]
```diff
@@ -1,63 +1,58 @@
-import React from "react"
-import PropTypes from "prop-types"
-import { createStoreObject } from "effector"
+// @flow
+import * as React from "react"
```
[the remark this answers is not in the document]
Comment (sergeysova) [reply src/features/cards/organisms/cards-list.js]: Flow likes star import for react

File: src/features/cards/organisms/cards-list.js (hunk 1 above)
Comment (sergeysova) [reply src/features/cards/organisms/cards-list.js]: In this case you will have array with empty elements

File: src/features/cards/organisms/skeleton-list.js (hunk 4 above)
Comment (sergeysova) [reply src/features/cards/organisms/skeleton-list.js]: Because eslint can't see default parameter

File: src/features/users/pages/user.js (hunk 7 above)
Comment (sergeysova) [reply src/features/users/pages/user.js]: For flowtype checking. Yes

File: src/features/cards/organisms/cards-list.js (hunk 1 above)
Comment (artalar) [reply src/features/cards/organisms/cards-list.js]: Array of `false` and needed elements, so what?

Comment (sergeysova) [reply src/features/cards/organisms/cards-list.js]: I think this is eager optimization

File: src/features/users/pages/user.js (hunk 6 above)
Comment (sergeysova) [reply src/features/users/pages/user.js]: This is not a final solution. After redesign, I will return to this code

**Pull request merged**